util/changelog.py

```diff
@@ -1,18 +1,35 @@
 import argparse
 import copy
 import sys
+import re
+from typing import Tuple
 
 """
 Changelog generation script, requires PAT with public_repo access, 
 see https://github.com/settings/tokens
 
-Caveats V20 and prior release tags are tips on their respective release branches
-If you try to use a start tag with one of these a full changelog will be generated
-since the commit wont appear in your iterations
+usage: changelog [-h] [-e END] [-m {full,incremental}] -p PAT [-r REPO] [-s START] [-t TAG]
```

Comment:
@zhyatt , needs you careful review on these changes. This script seemed to be broken in one of my runs. It didn't work for getting the changelogs that would be added to the RPMs. Where else is this script used on?

Reply:
As we discussed on our call, testing this command locally with `./util/changelog.py --start [hash] --end [hash] --pat [personal access token]` should help determine if there is an issue with the script or the way the CI is calling it. Determining the start hash is a bit tricky due to the way we branch for a release from develop.

To get the start hash, from the previous release branch (so for V23, this would be `git checkout releases/v22`), run this command: `diff -u <(git rev-list --first-parent "${1:-develop}") <(git rev-list --first-parent "${2:-HEAD}") | sed -ne "s/^ //p" | head -1`.

Reply:
https://git-scm.com/docs/git-merge-base
using with --octopus seems to do the same for finding the common ancestor

Reply:
BTW, I'll remove the change log generation from this PR and create new workflow for it.